src/node.js

```diff
@@ -388,17 +389,17 @@
       tickDone();
     }
 
+    function PObj(c) {
+      this.callback = c;
+      this.domain = process.domain || null;
+    }
+
     function nextTick(callback) {
       // on the way out, don't bother. it won't get fired anyway.
       if (process._exiting)
         return;
 
-      var obj = {
-        callback: callback,
-        domain: process.domain || null
-      };
-
-      nextTickQueue.push(obj);
+      nextTickQueue.push(new PObj(callback));
```

Comment:
Oh, I didn't say using a class isn't faster - I said using `var obj = {...}` and then `.push(obj)` isn't slower than using `.push({...})` after JIT. Although the fact using `new Obj` with a constructor is faster than a literal is indeed surprising.

Just for the sport - is doing `function PObj(){ return {...}; }` as fast as the inline literal version or as the constructor function version? I'd expect the latter.